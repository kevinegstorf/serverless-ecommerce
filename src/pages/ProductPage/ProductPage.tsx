import React, { Component } from 'react';
import Product from '../../components/Product/Product';

import config from "../../config";

export default class ProductsPage extends Component {
    state = {
        products: []
      }
      
      componentDidMount() {
        this.fetchProducts();
      }

    
      async fetchProducts(): Promise<void> { // Token returned from Stripe
        try {
            const res = await fetch(config.stripe.productsUrl, { // Backend API url
              method: 'GET'
            });
            const response = await res.json();
            const products = response.data;
        
            this.setState({
              products
            });
        } catch (error){
            console.log(error)
        }
      }
    
    
      render() {
        const {products} = this.state;
    
        const productList = products.map((product: any, index) => {
          return (
              <Product  
                id={product.id}
                key ={product.name}
                name={product.name}
                caption={product.caption}
                description={product.description}
                skus=''
                images={product.images} />
          );
        });
    
        return (
            <div id="products">
            <h1 className='testing'>Test</h1>
              {productList}
            </div>
        );
      }
    }