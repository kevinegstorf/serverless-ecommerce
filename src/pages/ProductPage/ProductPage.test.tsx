import React from 'react';
import { render } from 'react-testing-library';
import ProductPage from './ProductPage';

test('snapshot test', () => {
    const container = render(<ProductPage/>)
    expect(container).toMatchSnapshot();
});