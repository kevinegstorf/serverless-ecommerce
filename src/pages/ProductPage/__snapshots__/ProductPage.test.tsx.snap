// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="products"
      >
        <h1
          class="testing"
        >
          Test
        </h1>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="products"
    >
      <h1
        class="testing"
      >
        Test
      </h1>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
